neigh_linux.go

```diff
@@ -191,34 +191,30 @@ func NeighProxyList(linkIndex, family int) ([]Neigh, error) {
 	return pkgHandle.NeighProxyList(linkIndex, family)
 }
 
-// NeighListFiltered returns a list of neighbour entries filtered by link, ip family, flag and state.
-func NeighListFiltered(linkIndex, family, flags int, state uint16) ([]Neigh, error) {
-	return pkgHandle.NeighListFiltered(linkIndex, family, flags, state)
-}
-
 // NeighList returns a list of IP-MAC mappings in the system (ARP table).
 // Equivalent to: `ip neighbor show`.
 // The list can be filtered by link and ip family.
 func (h *Handle) NeighList(linkIndex, family int) ([]Neigh, error) {
-	return h.NeighListFiltered(linkIndex, family, 0, 0)
+	return h.NeighListExecute(Ndmsg{
+		Family: uint8(family),
+		Index:  uint32(linkIndex),
+	})
 }
 
 // NeighProxyList returns a list of neighbor proxies in the system.
 // Equivalent to: `ip neighbor show proxy`.
 // The list can be filtered by link, ip family.
 func (h *Handle) NeighProxyList(linkIndex, family int) ([]Neigh, error) {
-	return h.NeighListFiltered(linkIndex, family, NTF_PROXY, 0)
+	return h.NeighListExecute(Ndmsg{
+		Family: uint8(family),
+		Index:  uint32(linkIndex),
+		Flags:  NTF_PROXY,
+	})
 }
 
-// NeighListFiltered returns a list of neighbour entries filtered by link, ip family, flag and state.
-func (h *Handle) NeighListFiltered(linkIndex, family, flags int, state uint16) ([]Neigh, error) {
+// NeighListExecute returns a list of neighbour entries filtered by link, ip family, flag and state.
+func (h *Handle) NeighListExecute(msg Ndmsg) ([]Neigh, error) {
```

Comment:
Please also add a package function for this `func NeighListExecute(msg Ndmsg) ([]Neigh, error) {`

Reply:
done